src/commands/call/contract.rs

```diff
@@ -0,0 +1,148 @@
+	value: BalanceVariant<<DefaultEnvironment as Environment>::Balance>,
+	/// Maximum amount of gas to be used for this command.
+	/// If not specified will perform a dry-run to estimate the gas consumed for the
+	/// instantiation.
+	#[clap(name = "gas", long)]
+	gas_limit: Option<u64>,
+	/// Maximum proof size for this instantiation.
+	/// If not specified will perform a dry-run to estimate the proof size required.
+	#[clap(long)]
+	proof_size: Option<u64>,
+	/// Websocket endpoint of a node.
+	#[clap(name = "url", long, value_parser, default_value = "ws://localhost:9944")]
+	url: url::Url,
+	/// Secret key URI for the account deploying the contract.
+	///
+	/// e.g.
+	/// - for a dev account "//Alice"
+	/// - with a password "//Alice///SECRET_PASSWORD"
+	#[clap(name = "suri", long, short)]
+	suri: String,
+	/// Submit the extrinsic for on-chain execution.
+	#[clap(short('x'), long)]
+	execute: bool,
+}
+
+impl CallContractCommand {
+	pub(crate) async fn execute(&self) -> anyhow::Result<()> {
+		clear_screen()?;
+		intro(format!("{}: Calling a contract", style(" Pop CLI ").black().on_magenta()))?;
+		set_theme(Theme);
+
+		let token_metadata = TokenMetadata::query::<DefaultConfig>(&self.url).await?;
+		let call_exec = self.set_up_call(token_metadata.clone()).await?;
+
+		if !self.execute {
+			let mut spinner = cliclack::spinner();
+			spinner.start("Calling the contract...");
+			let call_dry_run_result = dry_run_call(&call_exec).await?;
+			log::info(format!("Result: {}", call_dry_run_result))?;
+			log::warning("Your call has not been executed.")?;
+			log::warning(format!(
+                    "To submit the transaction and execute the call on chain, add {} flag to the command.",
+                    "-x/--execute"
+            ))?;
+		} else {
+			let weight_limit;
+			if self.gas_limit.is_some() && self.proof_size.is_some() {
+				weight_limit =
+					Weight::from_parts(self.gas_limit.unwrap(), self.proof_size.unwrap());
+			} else {
+				let mut spinner = cliclack::spinner();
+				spinner.start("Doing a dry run to estimate the gas...");
+				weight_limit = match dry_run_gas_estimate_call(&call_exec).await {
+					Ok(w) => {
+						log::info(format!("Gas limit {:?}", w))?;
+						w
+					},
+					Err(e) => {
+						spinner.error(format!("{e}"));
+						outro_cancel("Deployment failed.")?;
+						return Ok(());
+					},
+				};
+			}
+			let mut spinner = cliclack::spinner();
+			spinner.start("Calling the contract...");
+
+			let call_result = call_smart_contract(call_exec, weight_limit, token_metadata)
+				.await
+				.map_err(|err| anyhow!("{} {}", "ERROR:", format!("{err:?}")))?;
+
+			log::info(call_result)?;
+		}
+
+		outro("Call completed successfully!")?;
+		Ok(())
+	}
+
+	async fn set_up_call(
+		&self,
+		token_metadata: TokenMetadata,
+	) -> anyhow::Result<CallExec<DefaultConfig, DefaultEnvironment, Keypair>> {
+		// If the user specify a path (not current directory) have to manually add Cargo.toml here
```

Comment:
"If the user specifies a path (which is not the current directory), it will have to manually add a `Cargo.toml` file. If not provided, `pop-cli` will ask the user for a specific path.